modules/builds-running-entitled-builds-with-sharedsecret-objects.adoc

```diff
@@ -0,0 +1,161 @@
+====
+
+.Procedure
+
+. Create a `SharedSecret` object instance with the cluster's entitlement secret by running the following command:
++
+[IMPORTANT]
+====
+You must have the cluster administrator permissions to create `SharedSecret` objects.
+====
++
+[source,yaml]
+----
+$ oc apply -f -<<EOF
+apiVersion: sharedresource.openshift.io/v1alpha1
+kind: SharedSecret
+metadata:
+  name: etc-pki-entitlement
+spec:
+  secretRef:
+    name: etc-pki-entitlement
+    namespace: openshift-config-managed
+EOF
+----
+
+. Create a `ClusterRole` to grant permission to access the `SharedSecret` object:
++
+[source,yaml]
+----
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: use-share-etc-pki-entitlement
+rules:
+  - apiGroups:
+      - sharedresource.openshift.io
+    resources:
+      - sharedsecrets
+    resourceNames:
+      - etc-pki-entitlement
+    verbs:
+      - use
+----
+
+. Create a `Role` and `RoleBinding` object that grants the Shared Resources CSI driver the permission to access the `SharedSecret` object:
++
+[source,yaml]
+----
+apiVersion: rbac.authorization.k8s.io/v1
+kind: Role
+metadata:
+  name: share-etc-pki-entitlement
+  namespace: openshift-config-managed
+rules:
+  - apiGroups:
+      - ""
+    resources:
+      - secrets
+    resourceNames:
+      - etc-pki-entitlement
+    verbs:
+      - get
+      - list
+      - watch
+----
++
+[source,yaml]
+----
+apiVersion: rbac.authorization.k8s.io/v1
+kind: RoleBinding
+metadata:
+  name: share-etc-pki-entitlement
+  namespace: openshift-config-managed
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: Role
+  name: share-etc-pki-entitlement
+subjects:
+  - kind: ServiceAccount
+    name: csi-driver-shared-resource
+    namespace: openshift-builds <1>
+----
++
+<1> Enter the name of the namespace where openshift-builds is installed.
+
+. Create a `RoleBinding` object for the ServiceAccount that consumes the `SharedSecret` in your current namespace:
++
+[source,yaml]
+----
+apiVersion: rbac.authorization.k8s.io/v1
+kind: RoleBinding
+metadata:
+  name: use-share-etc-pki-entitlement
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: use-share-etc-pki-entitlement
+subjects:
+  - kind: ServiceAccount
+    name: pipeline
+  - kind: ServiceAccount
+    name: builder
+----
++
+[NOTE]
+====
+The ServiceAccount that consumes the `SharedSecret` are created and managed by the OpenShift controllers.
+====
+
+. Mount the `SharedSecret` with relevant mount details using the `buildah` buildStrategy:
++
+[source,yaml]
+----
+$ oc apply -f -<<EOF
+apiVersion: shipwright.io/v1beta1
+kind: Build
+metadata:
+  name: buildah-rhel
+spec:
+  source:
+    type: Git
+    git:
+      url: <LOCATION_OF_OPERATOR_ORGANIZATION>
+      revision: entitled-builds
+    contextDir: rh-builds
```

Comment:
This is the url where the build's source code resides. We should use our sample for buildah:

```suggestion
    git:
      url: https://github.com/redhat-openshift-builds/samples
    contextDir: buildah-build
```